### PR TITLE
Optim

### PR DESCRIPTION
**Pull request by jschueller**
Only optim-related Fixes/enhancements, mostly moved from #2092:
- Enable new moead_gen algo from pagmo=>2.19
- Enable all Bonmin algos starting from bonmin>=1.8.9
- New setResultFromEvaluationHistory method to populate result from history as sometimes the best point returned is not the optimal / feasible
- Misc TNC/Cobyla bugs
- Catch exceptions in Bonmin/Ipopt callbacks (avoids segfaults)
- Fix invalid GLM optim bounds

### CODE REVIEW

**Comment by sofianehaddad** [comment]
@jschueller `dlib` has an `adam` solver. Any plan to parse it for openturns?

**Comment by jschueller** [comment]
the problem is it seems exposed with a different api than the rest of the optim algos, dedicated to the neural nets

**Review by regislebrun — APPROVED** [review]

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx [hunk 1]
```diff
@@ -209,7 +209,14 @@ void GeneralLinearModelAlgorithm::setCovarianceModel(const CovarianceModel & cov
     if (isScaleActive)
     {
       const Point inputSampleRange(inputSample_.getMax() - inputSample_.getMin());
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx:204]: wow, didnt know it even existed

**Comment by jschueller** [comment]
@sofianehaddad can we merge this ?

**Comment by sofianehaddad** [comment]
+1

**Comment by jschueller** [comment]
cool, thanks

**Comment by jschueller** [comment]
@m-balesdent these fixes should address the issues with non-feasible results you got on optimization problems

**Pull request merged**